include/cuco/static_map.cuh

```diff
@@ -590,14 +610,86 @@ class static_map {
+                                     Value empty_value_sentinel) noexcept
       : device_view_base{slots, capacity, empty_key_sentinel, empty_value_sentinel}
     {
     }
 
+    /**
+     * @brief Makes a copy of given `device_view` using non-owned memory.
+     *
+     * This function is intended to be used to create shared memory copies of small static maps, although global memory can be used as well.
+     *
+     * Example:
+     * @code{.cpp}
+     * template <typename MapType, int CAPACITY>
+     * __global__ void use_device_view(const typename MapType::device_view device_view,
+     *                                 map_key_t const* const keys_to_search,
+     *                                 map_value_t* const values_found,
+     *                                 const size_t number_of_elements)
+     * {
+     *     const size_t index = blockIdx.x * blockDim.x + threadIdx.x;
+     *
+     *     __shared__ typename MapType::pair_atomic_type sm_buffer[CAPACITY];
+     *
+     *     auto g = cg::this_thread_block();
+     *
+     *     const map_t::device_view sm_static_map = device_view.make_copy(g,
+     *                                                                    sm_buffer);
+     *
+     *     for (size_t i = g.thread_rank(); i < number_of_elements; i += g.size())
+     *     {
+     *         values_found[i] = sm_static_map.find(keys_to_search[i])->second;
+     *     }
+     * }
+     * @endcode
+     *
+     * @tparam CG The type of the cooperative thread group
+     * @param g The ooperative thread group used to copy the slots
+     * @param source_device_view `device_view` to copy from
+     * @param memory_to_use Array large enough to support `capacity` elements. Object does not take the ownership of the memory
+     * @return Copy of passed `device_view`
+     */
+    template <typename CG>
+    __device__ static device_view make_copy(CG g,
+                                            pair_atomic_type* const memory_to_use,
+                                            device_view source_device_view) noexcept
+    {
+#ifndef CUDART_VERSION
+#error CUDART_VERSION Undefined!
+#elif (CUDART_VERSION >= 11000)
+      __shared__ cuda::barrier<cuda::thread_scope::thread_scope_block> barrier;
+      if (g.thread_rank() == 0) {
+        init(&barrier, g.size());
+      }
+      g.sync();
+
+      cuda::memcpy_async(g,
+                         memory_to_use,
+                         source_device_view.get_slots(),
+                         sizeof(pair_atomic_type) * source_device_view.get_capacity(),
+                         barrier);
+
+      barrier.arrive_and_wait();
+#else
+      pair_atomic_type const* const slots_ptr = source_device_view.get_slots();
+      for (std::size_t i = g.thread_rank(); i < source_device_view.get_capacity(); i += g.size())
+      {
+        memory_to_use[i].first.store(slots_ptr[i].first.load());
+        memory_to_use[i].second.store(slots_ptr[i].second.load());
```

Comment:
If you don't specify a memory order argument, it defaults to sequential consistency, which is the strongest (and slowest) memory order. I think this is the most efficient way you can do this operation:
```suggestion
        new (&memory_to_use[i].first.) atomic_key_type{slots_ptr[i].first.load(cuda::memory_order_relaxed)};
        new (&memory_to_use[i].second) atomic_value_type{slots_ptr[i].second.load(cuda::memory_order_relaxed)};
```

This avoids the need to do a `store` operation, and uses a `load(cuda::memory_order_relaxed)`.

Reply:
You can see how we do the same thing in the initialization kernel here: https://github.com/NVIDIA/cuCollections/blob/9f378842de160fe80f607cdea0c4f6781bf979b9/include/cuco/detail/static_map_kernels.cuh#L37-L48

Reply:
Thanks, that makes sense. I still have to catch up on memory model.
done